packages/flame/test/effects/controllers/sine_effect_controller_test.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:math';
+
+import 'package:flame/effects.dart';
+import 'package:flame_test/flame_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('SineEffectController', () {
+    test('general properties', () {
+      final ec = SineEffectController(period: 1);
+      expect(ec.duration, 1);
+      expect(ec.started, true);
+      expect(ec.completed, false);
+      expect(ec.progress, 0);
+      expect(ec.isRandom, false);
+    });
+
+    test('progression', () {
+      final ec = SineEffectController(period: 3);
+      final expectedProgress = [
+        for (var i = 0; i <= 100; i++) sin(i * 0.01 * 2 * pi),
```

Comment:
Why not use `List.generate` instead of all these for loops inside of lists?

Reply:
yeah, in this case List.generate might be nicer